packages-content-model/roosterjs-content-model-editor/lib/modelApi/table/applyTableFormat.ts

```diff
@@ -178,71 +195,67 @@ function formatBorders(rows: ContentModelTableRow[], format: TableMetadataFormat
                     color: borderColor,
                 });
             });
-        });
-    });
-}
-
-function formatBackgroundColors(
-    rows: ContentModelTableRow[],
-    format: TableMetadataFormat,
-    bgColorOverrides: boolean[][]
-) {
-    const { hasBandedRows, hasBandedColumns, bgColorOdd, bgColorEven } = format;
 
-    rows.forEach((row, rowIndex) => {
-        row.cells.forEach((cell, colIndex) => {
-            if (!bgColorOverrides[rowIndex][colIndex]) {
+            // Format Background Color
+            if (!metaOverrides.bgColorOverrides[rowIndex][colIndex]) {
                 const color =
                     hasBandedRows || hasBandedColumns
                         ? (hasBandedColumns && colIndex % 2 != 0) ||
                           (hasBandedRows && rowIndex % 2 != 0)
                             ? bgColorOdd
                             : bgColorEven
-                        : bgColorEven;
+                        : bgColorEven; /* bgColorEven is the default color */
 
                 setTableCellBackgroundColor(cell, color);
             }
+
+            // Format Vertical Align
+            if (!metaOverrides.vAlignOverrides[rowIndex][colIndex]) {
+                cell.format.verticalAlign = format.verticalAlign;
+            }
         });
     });
 }
 
 function setFirstColumnFormat(
     rows: ContentModelTableRow[],
     format: Partial<TableMetadataFormat>,
-    bgColorOverrides: boolean[][]
+    metaOverrides: MetaOverrides
 ) {
-    rows.forEach((row, rowIndex) => {
-        row.cells.forEach((cell, cellIndex) => {
-            if (format.hasFirstColumn && cellIndex === 0) {
```

Comment:
why remove this check?

Reply:
It is not removed, but format.hasFirstColumn is constant so I moved it outside.

Reply:
But this changes the logic.

In original code, if format.hasFirstColumn is false, it will still execute line 228 (line 240 after change) since it is in "else" block, but after this change nothing will be executed.

Reply:
You are right, I did not notice.